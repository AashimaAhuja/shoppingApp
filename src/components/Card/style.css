.img-container > img {
  height: 250px;
  width: 250px;
}
.product-card {
  margin: 8px;
}
.img-container {
  position: relative;
}

.product-card label {
  display: none;
}

.product-card.sold label {
  display: block;
  font-size: 50px;
  color: #fff;
  position: absolute;
  left: 50%;
  top: 50%;
  transform: translate(-50%, -50%);
  z-index: 1;
}

.product-card.sold {
  opacity: 0.5;
}
.img-container > button {
  position: absolute;
  right: 5%;
  top: 5%;
}

.product-info > dt {
  margin: 0;
  line-height: 1.5rem;
}

.product-info > dt:nth-child(1),
.product-info > dt:last-child {
  font-weight: 700;
}
