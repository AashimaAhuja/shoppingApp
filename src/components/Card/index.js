import React from "react";
import "./style.css";

const Card = ({ product, onProductLike }) => {
  const { img, name, brand, price, size, sold, id, isLiked } = product;

  const toggleLikeBtn = () => {
    if (!sold) {
      onProductLike(id);
    }
  };

  return (
    <div className={`product-card ${sold ? "sold" : ""}`}>
      <div className="img-container">
        <img src={img} alt='card' />
        <label>SOLD</label>
        <button onClick={toggleLikeBtn}>{isLiked ? "Unlike" : "Like"}</button>
      </div>

      <dl className="product-info">
        <dt>{name}</dt>
        <dt>{brand}</dt>
        <dt>{size}</dt>
        <dt>{price}</dt>
      </dl>
    </div>
  );
};

export default Card;
