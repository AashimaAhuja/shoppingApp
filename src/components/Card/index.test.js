import { shallow } from 'enzyme';
import * as React from 'react';
import Card from './';

const defaultProps = {
    product: {
        img: 'imageUrl',
        name: 'Prod 1',
        brand: 'Brand 1',
        price: 30,
        size: 30,
        sold: false,
        id: 1,
        isLiked: false
    },
    onProductLike: jest.fn()
}

describe('<Card/> Tests', () => {
    it('should render Card Component', () => {
        const CardComponent = shallow(<Card product={{}} />)
        expect(CardComponent).toMatchSnapshot();
    });

    it('should render Card Component with default Props', () => {
        const wrapper = shallow(<Card {...defaultProps} />)
        expect(wrapper).toMatchSnapshot();
        const label = wrapper.find('.img-container label');
        expect(label.text()).toEqual('SOLD');
    });

    it('should render Card Component with product sold', () => {
        const props = {
            ...defaultProps,
            product: {
                ...defaultProps.product,
                sold: true
            }
        }
        const CardComponent = shallow(<Card {...props} />)
        expect(CardComponent).toMatchSnapshot();
    });
    it('should render Card Component with product not sold', () => {
        const props = {
            ...defaultProps,
            product: {
                ...defaultProps.product,
                sold: false
            }
        }
        const CardComponent = shallow(<Card {...props} />)
        expect(CardComponent).toMatchSnapshot();
    });

    it('should render Card Component with product liked', () => {
        const props = {
            ...defaultProps,
            product: {
                ...defaultProps.product,
                isLiked: true
            }
        }
        const wrapper = shallow(<Card {...props} />)
        expect(wrapper).toMatchSnapshot();
        const button = wrapper.find('button');
        expect(button.text()).toEqual('Unlike')
    });

    it('should render Card Component with product is not liked', () => {
        const props = {
            ...defaultProps,
            product: {
                ...defaultProps.product,
                isLiked: false
            }
        }
        const wrapper = shallow(<Card {...props} />)
        expect(wrapper).toMatchSnapshot();
        const button = wrapper.find('button');
        expect(button.text()).toEqual('Like')
    });

    it('should render Card Component with button click and product not sold', () => {
        const onProductLikeMock = jest.fn(this, [1]);
        const wrapper = shallow(<Card {...defaultProps} onProductLike={onProductLikeMock} />)
        expect(wrapper).toMatchSnapshot();
        const button = wrapper.find('button');
        button.simulate('click');
        expect(onProductLikeMock).toHaveBeenCalledTimes(1);
    });
    
    it('should render Card Component with button clikc and product sold', () => {
        const props = {
            ...defaultProps,
            product: {
                ...defaultProps.product,
                sold: true
            }
        }
        const onProductLikeMock = jest.fn();
        const wrapper = shallow(<Card {...props} onProductLike={onProductLikeMock} />)
        expect(wrapper).toMatchSnapshot();
        const button = wrapper.find('button');
        button.simulate('click');
        expect(onProductLikeMock).toHaveBeenCalledTimes(0);
    });
})
