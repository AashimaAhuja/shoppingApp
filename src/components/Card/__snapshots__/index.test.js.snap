// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card/> Tests should render Card Component 1`] = `
<div
  className="product-card "
>
  <div
    className="img-container"
  >
    <img
      alt="card"
    />
    <label>
      SOLD
    </label>
    <button
      onClick={[Function]}
    >
      Like
    </button>
  </div>
  <dl
    className="product-info"
  >
    <dt />
    <dt />
    <dt />
    <dt />
  </dl>
</div>
`;

exports[`<Card/> Tests should render Card Component with button click and product not sold 1`] = `
<div
  className="product-card "
>
  <div
    className="img-container"
  >
    <img
      alt="card"
      src="imageUrl"
    />
    <label>
      SOLD
    </label>
    <button
      onClick={[Function]}
    >
      Like
    </button>
  </div>
  <dl
    className="product-info"
  >
    <dt>
      Prod 1
    </dt>
    <dt>
      Brand 1
    </dt>
    <dt>
      30
    </dt>
    <dt>
      30
    </dt>
  </dl>
</div>
`;

exports[`<Card/> Tests should render Card Component with button clikc and product sold 1`] = `
<div
  className="product-card sold"
>
  <div
    className="img-container"
  >
    <img
      alt="card"
      src="imageUrl"
    />
    <label>
      SOLD
    </label>
    <button
      onClick={[Function]}
    >
      Like
    </button>
  </div>
  <dl
    className="product-info"
  >
    <dt>
      Prod 1
    </dt>
    <dt>
      Brand 1
    </dt>
    <dt>
      30
    </dt>
    <dt>
      30
    </dt>
  </dl>
</div>
`;

exports[`<Card/> Tests should render Card Component with default Props 1`] = `
<div
  className="product-card "
>
  <div
    className="img-container"
  >
    <img
      alt="card"
      src="imageUrl"
    />
    <label>
      SOLD
    </label>
    <button
      onClick={[Function]}
    >
      Like
    </button>
  </div>
  <dl
    className="product-info"
  >
    <dt>
      Prod 1
    </dt>
    <dt>
      Brand 1
    </dt>
    <dt>
      30
    </dt>
    <dt>
      30
    </dt>
  </dl>
</div>
`;

exports[`<Card/> Tests should render Card Component with product is not liked 1`] = `
<div
  className="product-card "
>
  <div
    className="img-container"
  >
    <img
      alt="card"
      src="imageUrl"
    />
    <label>
      SOLD
    </label>
    <button
      onClick={[Function]}
    >
      Like
    </button>
  </div>
  <dl
    className="product-info"
  >
    <dt>
      Prod 1
    </dt>
    <dt>
      Brand 1
    </dt>
    <dt>
      30
    </dt>
    <dt>
      30
    </dt>
  </dl>
</div>
`;

exports[`<Card/> Tests should render Card Component with product liked 1`] = `
<div
  className="product-card "
>
  <div
    className="img-container"
  >
    <img
      alt="card"
      src="imageUrl"
    />
    <label>
      SOLD
    </label>
    <button
      onClick={[Function]}
    >
      Unlike
    </button>
  </div>
  <dl
    className="product-info"
  >
    <dt>
      Prod 1
    </dt>
    <dt>
      Brand 1
    </dt>
    <dt>
      30
    </dt>
    <dt>
      30
    </dt>
  </dl>
</div>
`;

exports[`<Card/> Tests should render Card Component with product not sold 1`] = `
<div
  className="product-card "
>
  <div
    className="img-container"
  >
    <img
      alt="card"
      src="imageUrl"
    />
    <label>
      SOLD
    </label>
    <button
      onClick={[Function]}
    >
      Like
    </button>
  </div>
  <dl
    className="product-info"
  >
    <dt>
      Prod 1
    </dt>
    <dt>
      Brand 1
    </dt>
    <dt>
      30
    </dt>
    <dt>
      30
    </dt>
  </dl>
</div>
`;

exports[`<Card/> Tests should render Card Component with product sold 1`] = `
<div
  className="product-card sold"
>
  <div
    className="img-container"
  >
    <img
      alt="card"
      src="imageUrl"
    />
    <label>
      SOLD
    </label>
    <button
      onClick={[Function]}
    >
      Like
    </button>
  </div>
  <dl
    className="product-info"
  >
    <dt>
      Prod 1
    </dt>
    <dt>
      Brand 1
    </dt>
    <dt>
      30
    </dt>
    <dt>
      30
    </dt>
  </dl>
</div>
`;
